app/lib/common/pages/medications/cubit.dart

```diff
@@ -13,17 +15,51 @@ class MedicationsCubit extends Cubit<MedicationsState> {
   final int _id;
 
   Future<void> loadMedications() async {
-    final requestUri = annotationServerUrl.replace(
-      path: 'api/v1/medications/$_id',
-    );
     emit(MedicationsState.loading());
+    final response = await sendWithRandomRequests();
+    if (response == null){
+      emit(MedicationsState.error());
+    }
+    else {
+      final medication = medicationWithGuidelinesFromHTTPResponse(response);
+      emit(MedicationsState.loaded(medication));
+    }
+  }
+
+  Future<http.Response> sendRequest(Uri requestUri) async {
     final response = await http.get(requestUri);
     if (response.statusCode != 200) {
       emit(MedicationsState.error());
-      return;
     }
-    final medication = medicationWithGuidelinesFromHTTPResponse(response);
-    emit(MedicationsState.loaded(medication));
+    return response;
+  }
+
+  Future<http.Response?> sendWithRandomRequests() async {
+    final requestIdsUri = annotationServerUrl.replace(
+      path: 'api/v1/medications/ids',
+    );
+    final idsResponse = await http.get(requestIdsUri);
+    final idsMapList = jsonDecode(idsResponse.body) as List<dynamic>;
+
+    final idList = [];
+    for (final element in idsMapList) {
+      idList.add(element['id']);
+    }
+
+    final randomIds = idList.sample(2);
+    randomIds.add(_id);
+    randomIds.shuffle();
+    http.Response? response;
+    for (final id in randomIds) {
+      final requestMedicationUri = annotationServerUrl.replace(
+        path: 'api/v1/medications/$id',
+      );
+      final tempResponse = await sendRequest(requestMedicationUri);
```

Comment:
```suggestion
      final tempResponse = await http.get(requestMedicationUri);
      if (tempResponse!.statusCode != 200) {
        emit(MedicationsState.error);
        return null;
      }
```

Reply:
the "!" got me an `dart[unnecessary_non_null_assertion]` so I left it out